src/solvers/lowering/byte_operators.cpp

```diff
@@ -1106,6 +1106,7 @@ exprt lower_byte_extract(const byte_extract_exprt &src, const namespacet &ns)
     if(element_bits.has_value() && *element_bits >= 1 && *element_bits % 8 == 0)
     {
       auto num_elements = numeric_cast<std::size_t>(array_type.size());
+      // XXX: This can't be right -- unpacked is a byte array, whereas array_type may have larger elements
```

Comment:
Agreed, this could result in inconsistent expressions being generated. We should really stick with the `else` branch here (which AFAIK was added much later). Can you amend your PR to remove this `if` statement here?

Reply:
Nevermind, I have found a test exercising this path. I'll merge this one and will create a follow-up PR.